### PR TITLE
capz: add presubmit job for windows tests

### PR DESCRIPTION
**Pull request by cpanato**
will not run by default and it is not required as well

/assign @CecileRobertMichon @nader-ziada @devigned

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 1]
```diff
@@ -91,6 +91,37 @@ presubmits:
...
+      preset-azure-anonymous-pull: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20201208-8b354d9-1.18
+        command:
+        - "runner.sh"
+        - "./scripts/ci-e2e.sh"
+        env:
+          - name: GINKGO_FOCUS
+            value: "Workload cluster creation"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:113]: We also need to skip windows tests in the linux e2e. How about changing this to add a Focus on Windows and changing the `pull-cluster-api-provider-azure-e2e` to Skip windows?

Comment (nader-ziada) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:113]: if we have two different contexts; `linux workload cluster` and `windows workload cluster`, the appropriate job can just set the focus to that

Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml]: makes sense, done

Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml]: i prefer that actually

**Review by CecileRobertMichon — COMMENTED** [review]
/assign @jsturtevant

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ presubmits:
           - name: GINKGO_FOCUS
             value: "Workload cluster creation"
           - name: GINKGO_SKIP
-            value: "Creating a GPU-enabled cluster"
+            value: "Creating a GPU-enabled cluster|Creating a Windows enabled VMSS cluster"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:53]: There are actually two different contexts for Windows right now https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1036/files#diff-93ce65d81f5c1510135f6035d43f678bf9b5962b8e4ee45105e3345a1852efc6R354

How can we make this more generic so it doesn't break in the future if more windows tests are added?

Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:53]: Could we use a well known set of test of tags?  Something like `[presubmitt optiontal]` or more specific like `[Windows]`? Are there any like this in use in Cluster API anywhere?

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 3]
```diff
@@ -91,6 +91,37 @@ presubmits:
...
+      preset-azure-anonymous-pull: "true"
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20201208-8b354d9-1.18
+        command:
+        - "runner.sh"
+        - "./scripts/ci-e2e.sh"
+        env:
+          - name: GINKGO_FOCUS
+            value: "Creating a Windows enabled VMSS cluster"
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:113]: same here, we have two sets of tests right now (VMSS and VMAS)

**Comment by jsturtevant** [comment]
Do we have the same concerns around length of time for the periodics? Should those be split into separate jobs as well?

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml (hunk 2 above)
Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:53]: we can make a regex that filters the `Windows` keyword

or we create another `Describe` and add all windows tests there

Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml:53]: Using `Windows` works for me

Comment (cpanato) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml]: updated PTAL
@CecileRobertMichon @jsturtevant

**Comment by cpanato** [comment]
@jsturtevant will add the periodic when we merge the PR and check if the presubmit works :)

**Review by devigned — APPROVED** [review]
lgtm

**Pull request merged**